src/java/org/apache/cassandra/auth/DataResource.java

```diff
@@ -249,7 +250,8 @@ public boolean exists()
                 return true;
             case KEYSPACE:
             case ALL_TABLES:
-                return Schema.instance.getKeyspaces().contains(keyspace);
+                return Schema.instance.getKeyspaces().contains(keyspace) ||
+                        VirtualKeyspaceRegistry.instance.getKeyspaces().contains(keyspace);
```

Comment:
can we use `SchemaConstants.isVirtualSystemKeyspace(keyspace)` instead?
```suggestion
                        SchemaConstants.isVirtualSystemKeyspace(keyspace);
```

Reply:
Just going over the code again, and it seems like SchemaConstants.isVirtualSystemKeyspace is not a good fit, because it only takes into account __virtual__ table system keyspace names. So I'm walking back on this suggestion.

Reply:
I spoke offline with @Maxwell-Guo and discussed about whether `SchemaConstants.isVirtualSystemKeyspace` is a good fit or not. While `VirtualKeyspaceRegistry` offers a dynamic approach to checking whether a virtual keyspace exists or not, `SchemaConstants.isVirtualSystemKeyspace` has statically defined virtual keyspace names. Is there a scenario where these two diverge? And if they do not diverge can we use `SchemaConstants.isVirtualSystemKeyspace` instead?